### PR TITLE
msa tests 

### PR DESCRIPTION
**Pull request by jennyf19**

### CODE REVIEW

File: src/client/Microsoft.Identity.Client/MsalError.cs [hunk 1]
```diff
@@ -466,7 +466,7 @@ public static class MsalError
         /// <summary>
         /// Non HTTPS redirects are not supported
         /// <para>What happens?</para>This error happens when you have registered a non-HTTPS redirect URI for the
-        /// public client application other than <c>urn:ietf:wg:oauth:2.0:oob</c>
+        /// public client application other than <c>https://login.microsoftonline.com/common/oauth2/nativeclient</c>
```
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/MsalError.cs]: This is not true.

File: src/client/Microsoft.Identity.Client/MigrationAid/MigrationAid.cs [hunk 2]
```diff
@@ -144,7 +144,7 @@ public partial interface IClientApplicationBase
         /// This redirect URI needs to be registered in the app registration (https://aka.ms/msal-net-register-app)
         /// In MSAL.NET, <see cref="T:PublicClientApplication"/> define the following default RedirectUri values:
         /// <list type="bullet">
-        /// <item><description><c>urn:ietf:wg:oauth:2.0:oob</c> for desktop (.NET Framework and .NET Core) applications</description></item>
+        /// <item><description><c>https://login.microsoftonline.com/common/oauth2/nativeclient</c> for desktop (.NET Framework and .NET Core) applications</description></item>
```
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/MigrationAid/MigrationAid.cs]: Please create a separate PR for these doc changes.

File: tests/Microsoft.Identity.Test.Common/TestConstants.cs [hunk 3]
```diff
@@ -82,7 +82,7 @@ public static SortedSet<string> s_scope
         public const string Name = "First Last";
         public const string Claims = "claim1claim2";
         public const string DisplayableId = "displayable@id.com";
-        public const string RedirectUri = "urn:ietf:wg:oauth:2.0:oob";
+        public const string RedirectUri = "https://login.microsoftonline.com/common/oauth2/nativeclient";
```
Comment (bgavrilMS) [inline tests/Microsoft.Identity.Test.Common/TestConstants.cs]: I'd also think this is a separate change.

File: tests/Microsoft.Identity.Test.Integration.net45/HeadlessTests/SilentAuthTests.cs [hunk 4]
```diff
@@ -18,6 +18,10 @@ public class SilentAuthTests
     {
         private static readonly string[] s_scopes = { "User.Read" };
 
+        /// <summary>
+        /// Initialized by MSTest (do not make private or readonly)
+        /// </summary>
+        public TestContext TestContext { get; set; }
```
Comment (bgavrilMS) [inline tests/Microsoft.Identity.Test.Integration.net45/HeadlessTests/SilentAuthTests.cs]: Why do you need the test context?

File: tests/Microsoft.Identity.Test.Integration.net45/HeadlessTests/UsernamePasswordIntegrationTests.cs [hunk 5]
```diff
@@ -76,6 +76,39 @@ public async Task AcquireTokenFromAdfsUsernamePasswordAsync()
...
+            var msalPublicClient = PublicClientApplicationBuilder.Create(labResponse.App.AppId).WithAuthority(_authority).Build();
+
+            try
+            {
+                var result = await msalPublicClient
+                    .AcquireTokenByUsernamePassword(s_scopes, user.Upn, securePassword)
+                    .ExecuteAsync(CancellationToken.None)
+                    .ConfigureAwait(false);
+            }
+            catch (MsalServiceException ex)
```
Comment (bgavrilMS) [inline tests/Microsoft.Identity.Test.Integration.net45/HeadlessTests/UsernamePasswordIntegrationTests.cs]: This is good,  but there's a helper that we tend to use called `AssertException` that pretty much wrapps the try / catch for you.

File: tests/Microsoft.Identity.Test.Integration.net45/SeleniumTests/SSHCertificateTests.cs [hunk 6]
```diff
@@ -73,12 +77,20 @@ public async Task Interactive_SSHCert_Async()
             await MsalAssert.AssertSingleAccountAsync(labResponse, pca, result).ConfigureAwait(false);
         }
 
+
+        [TestMethod]
+        public async Task Interactive_SSHCert_MsaUser_Async()
```
Comment (bgavrilMS) [inline tests/Microsoft.Identity.Test.Integration.net45/SeleniumTests/SSHCertificateTests.cs]: I don't think MSA can issue SSH certs, but it's a good test :)

File: tests/Microsoft.Identity.Test.Integration.net45/HeadlessTests/SilentAuthTests.cs (hunk 4 above)
Comment (jennyf19) [reply tests/Microsoft.Identity.Test.Integration.net45/HeadlessTests/SilentAuthTests.cs]: i need it for the device code tst..which i've now added

---
In reply to: [348463615](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1511#discussion_r348463615) [](ancestors = 348463615)

File: tests/Microsoft.Identity.Test.Integration.net45/SeleniumTests/SSHCertificateTests.cs (hunk 6 above)
Comment (jennyf19) [reply tests/Microsoft.Identity.Test.Integration.net45/SeleniumTests/SSHCertificateTests.cs]: remvoed. thanks

---
In reply to: [348464362](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1511#discussion_r348464362) [](ancestors = 348464362)

File: tests/Microsoft.Identity.Test.Integration.net45/HeadlessTests/UsernamePasswordIntegrationTests.cs (hunk 5 above)
Comment (jennyf19) [reply tests/Microsoft.Identity.Test.Integration.net45/HeadlessTests/UsernamePasswordIntegrationTests.cs]: sweet. thanks

---
In reply to: [348463937](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1511#discussion_r348463937) [](ancestors = 348463937)

File: src/client/Microsoft.Identity.Client/AppConfig/ApplicationOptions.cs [hunk 7]
```diff
@@ -57,7 +57,7 @@ public abstract class ApplicationOptions
         /// This redirect URI needs to be registered in the app registration (https://aka.ms/msal-net-register-app).
         /// In MSAL.NET, <c>IPublicClientApplication</c> defines the following default RedirectUri values:
         /// <list type="bullet">
-        /// <item><description><c>urn:ietf:wg:oauth:2.0:oob</c> for desktop (.NET Framework and .NET Core) applications</description></item>
+        /// <item><description><c>https://login.microsoftonline.com/common/oauth2/nativeclient</c> for desktop (.NET Framework and .NET Core) applications</description></item>
```
Comment (bgavrilMS) [inline src/client/Microsoft.Identity.Client/AppConfig/ApplicationOptions.cs]: I still think these chagnes need to be in a separate commit.

File: tests/Microsoft.Identity.Test.Integration.net45/SeleniumTests/FociTests.cs [hunk 8]
```diff
@@ -49,7 +49,7 @@ public void TestInitialize()
         /// <remarks>The FOCI flag does not appear in the U/P flow, an interactive flow is required. Interactive flow
         /// cannot be automated because http://localhost cannot currently be added to the family apps</remarks>
         [TestMethod]
-        public async Task FociSingInSignOutAsync()
+        public async Task FociSignInSignOutAsync()
```
Comment (bgavrilMS) [inline tests/Microsoft.Identity.Test.Integration.net45/SeleniumTests/FociTests.cs]: A FOCI + MSA would be great

File: src/client/Microsoft.Identity.Client/AppConfig/ApplicationOptions.cs (hunk 7 above)
Comment (jennyf19) [reply src/client/Microsoft.Identity.Client/AppConfig/ApplicationOptions.cs]: i missed this one

**Comment by jennyf19** [comment]
we discussed offline...foci w/msa is not working w/one of the clientIds. i think they need to enable msa users

---
In reply to: [321082254](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1511#issuecomment-321082254) [](ancestors = 321082254)

**Comment by jennyf19** [comment]
i'll do a separate PR for those. Is that the only pending comment? 

---
In reply to: [321102190](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1511#issuecomment-321102190) [](ancestors = 321102190)

**Comment by bgavrilMS** [comment]
Yes, please commit when you are ready.
